### PR TITLE
Bump to conda 23.11.0 (with menuinst v2)

### PR DESCRIPTION
**Pull request by jaimergp**
### Description

Bump to latest `conda` and `conda-libmamba-solver`. Notably, this includes changes in how `menuinst` is called.

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

**Comment by jaimergp** [comment]
Failures should be fixed when https://github.com/conda/conda/pull/11882 is merged and released. In the meantime we can patch conda-standalone to include that, or just wait.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -29,32 +29,25 @@ jobs:
...
-          use-mamba: true
           run-post: false
-      - name: Install conda-libmamba-solver
+
+      - name: Install dependencies
         shell: bash -el {0}
-        run: |
-          mamba install "conda>=22.11.1" conda-libmamba-solver conda-build anaconda-client
-          conda update --all --solver=libmamba
+        run: conda install -y conda-build=3.27 anaconda-client
```
Comment (jaimergp) [inline .github/workflows/tests.yml:46]: To be unpinned once conda-build 3.28.2 is out.

File: src/entry_point.py [hunk 2]
```diff
@@ -212,10 +199,6 @@ def _constructor_subcommand():
     # when called with --make-menus and no package names, the value is an empty list
     # hence the explicit check for None
     elif (args.make_menus is not None) or args.rm_menus:
-        if sys.platform != "win32":
-            raise NotImplementedError(
-                "Menu creation and removal is only supported on Windows"
-            )
```
Comment (jezdez) [inline src/entry_point.py:218]: 💥

File: tests/test_main.py [hunk 3]
```diff
@@ -6,15 +6,40 @@
...
-    return subprocess.run([CONDA_EXE, *args], **kwargs)
+def run_conda(*args, **kwargs) -> subprocess.CompletedProcess:
+    check = kwargs.pop("check", False)
+    p = subprocess.run([CONDA_EXE, *args], **kwargs)
+    if check:
+        if kwargs.get("capture_output") and p.returncode:
+            print(p.stdout)
+            print(p.stderr, file=sys.stderr)
+        p.check_returncode()
+    return p
```
Comment (jezdez) [inline tests/test_main.py:26]:
```suggestion
    process = subprocess.run([CONDA_EXE, *args], **kwargs)
    if check:
        if kwargs.get("capture_output") and process.returncode:
            print(process.stdout)
            print(process.stderr, file=sys.stderr)
        process.check_returncode()
    return process
```

File: .github/workflows/tests.yml (hunk 1 above)
Comment (jezdez) [reply .github/workflows/tests.yml:46]: Do we need to wait for it?

**Review by jezdez — APPROVED** [review]

File: .github/workflows/tests.yml (hunk 1 above)
Comment (jaimergp) [reply .github/workflows/tests.yml:46]: No need. We can also do 

```suggestion
        run: conda install -y "conda-build!=3.28.0,!=3.28.1" anaconda-client
```

**Review by jezdez — APPROVED** [review]

**Pull request merged**